torch/_prims_common/__init__.py

```diff
@@ -476,33 +489,49 @@ class K(NamedTuple):
         stride: int
 
         def __lt__(self, other):
-            return guard_size_oblivious(self.stride < other.stride)
-
-        def __gt__(self, other):
-            return guard_size_oblivious(self.stride > other.stride)
+            # for backed symbols, this is practically a < operation
+            # for unbacked, we return True if < is statically known,
+            # then try to answer this symbolically, with stride ordering semantics
+            # (e.g. u0 < u0 is False, u0 < u1 is False with no axioms, u0 < 2 * u0 is True)
+            return (
+                guard_or_false(
+                    self.stride < other.stride
+                )  # checks statically known inequality
+                or (
+                    (
+                        guard_or_false(self.stride == 0)
+                        or guard_or_false(other.stride % self.stride == 0)
+                    )
+                    and guard_or_true(self.stride != other.stride)
+                )  # checks symbolic inequality (e.g. u0 < 2048 * u0)
+            )
 
-        def __le__(self, other):
-            return guard_size_oblivious(self.stride <= other.stride)
+    lengths_and_strides = sorted(map(K, sizes, strides))
 
-        def __ge__(self, other):
-            return guard_size_oblivious(self.stride >= other.stride)
+    # verify that sorted order was imposed (checks the "non-overlapping condition")
+    for i, j in zip(lengths_and_strides[:-1], lengths_and_strides[1:]):
+        if guard_or_false(i.stride == 0) or guard_or_false(j.stride % i.stride != 0):
+            return False
 
-        def __eq__(self, other):
-            return guard_size_oblivious(self.stride == other.stride)
+    # verify actual strides match the expected (composed sizes)
+    sizes = [x.size for x in lengths_and_strides][::-1]
+    strides = [x.stride for x in lengths_and_strides][::-1]
+    return check_contiguous_sizes_strides(sizes, strides, false_if_dde=True)
 
-    lengths_and_strides = sorted(map(K, a.shape, a.stride()))
 
-    expected_stride = 1
-    for length, stride in lengths_and_strides:
-        if guard_or_false(length == 1):
-            continue
+def is_non_overlapping_and_dense(a: Tensor) -> bool:
```

Comment:
is this for BC compatibility?
In my mind calling is_non_overlapping_and_dense should throw DDE
calling _is_non_overlapping_and_dense_or_false should not
now this is sort of alias for _is_non_overlapping_and_dense_or_false
meaning unbacked semantics are implicit to users calling is_non_overlapping_and_dense


maybe its ok